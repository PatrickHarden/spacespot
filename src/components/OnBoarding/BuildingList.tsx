import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import styled from 'styled-components'
import { useIntl } from 'react-intl'
import dashboardSelectors from 'services/dashboard/selectors'
import actions from 'services/onboarding/actions'
import { TranslatePropSpace } from 'services/space'
import Heading4 from 'components/common/Heading4'
import HTMLContent from 'components/common/HTMLContent'
import DOMPurify from 'dompurify'

export const BuildingBox = styled.div`
  box-sizing: border-box;
  border: 1px solid #dddddd;
  border-radius: 2px;
  border-color:#dddddd
  background-color: #f4f4f4;
  padding: 0 0 20px 0;
  color: #000000;
  font-weight: 300;
  line-height: 21px;
  text-align: center;
  margin-top: 20px;
  min-width: 350px;
  max-width: 407px;
  cursor: pointer;
  :hover {
    box-sizing: border-box;
    border: 1px solid #6CB9D5;
    box-shadow: 0 1px 4px 0 #6CB9D5;
  }

  @media(max-width: 480px){
    min-width: 100%;
  }
`

const BuildingHeading = styled.h2`
  color: black;
  font-size: 16px;
  line-height: 21px;
  font-weight: 500;
  text-align: left;
  margin: 10.5px 10.5px 0 10.5px;
`

const BuildingDescription = styled.h3`
  color: #828286;
  font-size: 16px;
  line-height: 18px;
  font-weight: 400;
  text-align: left;
  margin: 10.5px 10.5px 0 10.5px;
`

const BuildingList = () => {
  const dispatch = useDispatch()
  const buildings = useSelector(dashboardSelectors.selectDashboardBuildings)
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const selectBuilding = (selected: TranslatePropSpace) => () => {
    dispatch(actions.setBuildingId(selected.key))
    dispatch(actions.setName(selected.name))
    dispatch(actions.setDescription(selected.description))
    dispatch(actions.setLocation(selected.location || ''))
    dispatch(actions.setAddress(selected.address))
    dispatch(actions.setPostCode(selected.postCode))
    dispatch(actions.setCity(selected.city))
    dispatch(actions.setCountry(selected.country))
    dispatch(actions.setLatLng(selected.latLng))
    dispatch(actions.setAspects(selected.aspects))
    dispatch(actions.goToSpace('0'))
  }

  return (
    <>
      <Heading4>{t('ONBOARDING_EXISTING_SUBHEADER')}</Heading4>
      {buildings.map(building => (
        <BuildingBox key={building.key} onClick={selectBuilding(building)}>
          <BuildingHeading>{building.name}</BuildingHeading>
          <BuildingDescription>
            <HTMLContent
              html={DOMPurify.sanitize(
                (building.location || '').split(/\s+/).length > 8
                  ? (building.location || '')
                      .split(/\s+/)
                      .slice(0, 8)
                      .join(' ') + ' ...'
                  : building.location || '',
              )}
            />
          </BuildingDescription>
        </BuildingBox>
      ))}
    </>
  )
}

export default BuildingList
