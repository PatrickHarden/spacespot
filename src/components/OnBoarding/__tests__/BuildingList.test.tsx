import React from 'react'
import { Provider } from 'react-redux'
import { createMockStore } from 'redux-test-utils'
import { act } from 'react-dom/test-utils'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'

import GBPlus473580 from 'services/space/__mocks__/GB-Plus-473580.json'
import BuildingList, { BuildingBox } from '../BuildingList'

const convertJsonToSpace = (space: any) => ({
  ...space,
})

jest.mock('react-intl', () => ({
  ...jest.requireActual('react-intl'),
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

it('renders without crashing ', async () => {
  const spaces = [convertJsonToSpace(GBPlus473580).Document]
  const state = {
    dashboard: {
      buildings: spaces,
    },
  }
  const store: any = createMockStore(state)
  const wrapper = mount(
    <Provider store={store}>
      <BuildingList />
    </Provider>,
  )
  expect(mountToJson(wrapper)).toMatchSnapshot()
  const building = wrapper.find(BuildingBox)
  act(() => {
    building.simulate('click')
    wrapper.update()
  })
})

it('renders initial state', async () => {
  const state = {
    space: {},
  }
  const store: any = createMockStore(state)
  const wrapper = mount(
    <Provider store={store}>
      <BuildingList />
    </Provider>,
  )
  expect(mountToJson(wrapper)).toMatchSnapshot()
})
