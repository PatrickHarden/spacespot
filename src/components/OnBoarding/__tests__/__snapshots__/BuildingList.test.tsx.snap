// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders initial state 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <BuildingList>
    <styled.h4>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "iNjYGZ",
              "rules": Array [
                "
  color: #000000;
  font-size: 18px;
  font-weight: 500;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
",
              ],
            },
            "displayName": "styled.h4",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "h4",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <h4
          className="sc-bdVaJa iNjYGZ"
        >
          ONBOARDING_EXISTING_SUBHEADER
        </h4>
      </StyledComponent>
    </styled.h4>
  </BuildingList>
</Provider>
`;

exports[`renders without crashing  1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <BuildingList>
    <styled.h4>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "iNjYGZ",
              "rules": Array [
                "
  color: #000000;
  font-size: 18px;
  font-weight: 500;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
",
              ],
            },
            "displayName": "styled.h4",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "h4",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <h4
          className="sc-bdVaJa iNjYGZ"
        >
          ONBOARDING_EXISTING_SUBHEADER
        </h4>
      </StyledComponent>
    </styled.h4>
    <styled.div
      key="GB-Plus-473580"
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "lastClassName": "iSqFJV",
              "rules": Array [
                "
  box-sizing: border-box;
  border: 1px solid #dddddd;
  border-radius: 2px;
  border-color:#dddddd
  background-color: #f4f4f4;
  padding: 0 0 20px 0;
  color: #000000;
  font-weight: 300;
  line-height: 21px;
  text-align: center;
  margin-top: 20px;
  min-width: 350px;
  max-width: 407px;
  cursor: pointer;
  :hover {
    box-sizing: border-box;
    border: 1px solid #6CB9D5;
    box-shadow: 0 1px 4px 0 #6CB9D5;
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
      >
        <div
          className="sc-bwzfXH iSqFJV"
          onClick={[Function]}
        >
          <styled.h2>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": false,
                    "lastClassName": "diEYjS",
                    "rules": Array [
                      "
  color: black;
  font-size: 16px;
  line-height: 21px;
  font-weight: 500;
  text-align: left;
  margin: 10.5px 10.5px 0 10.5px;
",
                    ],
                  },
                  "displayName": "styled.h2",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "h2",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <h2
                className="sc-htpNat diEYjS"
              >
                Newly Refurbished 6th Floor Office Space Available
              </h2>
            </StyledComponent>
          </styled.h2>
          <styled.h3>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bxivhb",
                    "isStatic": false,
                    "lastClassName": "deSBxm",
                    "rules": Array [
                      "
  color: #828286;
  font-size: 16px;
  line-height: 18px;
  font-weight: 400;
  text-align: left;
  margin: 10.5px 10.5px 0 10.5px;
",
                    ],
                  },
                  "displayName": "styled.h3",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bxivhb",
                  "target": "h3",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <h3
                className="sc-bxivhb deSBxm"
              >
                <HTMLContent
                  html=""
                >
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "",
                      }
                    }
                  />
                </HTMLContent>
              </h3>
            </StyledComponent>
          </styled.h3>
        </div>
      </StyledComponent>
    </styled.div>
  </BuildingList>
</Provider>
`;
